types/src/v0/impls/state.rs

```diff
@@ -253,130 +303,329 @@ impl ValidatedState {
         Ok(())
     }
 }
+/// Block Proposal to be verified and applied.
+#[derive(Debug)]
+pub(crate) struct Proposal<'a> {
+    header: &'a Header,
+    block_size: u32,
```

Comment:
There is also a Proposal type in hotshot that we use in this repo, so I think we can use a different name here. `ProposalValidationInput`? A bit of a mouthful :roll_eyes:

Reply:
:thinking: